tests/basic/tests.py

```diff
@@ -164,10 +164,11 @@ def test_queryset_delete_removes_all_items_in_that_queryset(self):
             ["<Article: Amazing article>",
              "<Article: An article>",
              "<Article: Article One>",
-             "<Article: Boring article>"]
+             "<Article: Boring article>"],
+            repr,
```

Comment:
@felixxm  For this case, I can create the objects and put them on an array and then use them on assertion. like:

```diff
--- a/tests/basic/tests.py
+++ b/tests/basic/tests.py
@@ -157,18 +157,20 @@ class ModelTest(TestCase):
         headlines = [
             'An article', 'Article One', 'Amazing article', 'Boring article']
         some_pub_date = datetime(2014, 5, 16, 12, 1)
+        articles = []
         for headline in headlines:
-            Article(headline=headline, pub_date=some_pub_date).save()
+            articles.append(Article.objects.create(headline=headline, pub_date=some_pub_date))
         self.assertQuerysetEqual(
             Article.objects.all().order_by('headline'),
-            ["<Article: Amazing article>",
-             "<Article: An article>",
-             "<Article: Article One>",
-             "<Article: Boring article>"],
-            repr,
+            [
+                articles[2],
+                articles[0],
+                articles[1],
+                articles[3],
+            ],
         )
         Article.objects.filter(headline__startswith='A').delete()
-        self.assertQuerysetEqual(Article.objects.all().order_by('headline'), ["<Article: Boring article>"], repr)
+        self.assertQuerysetEqual(Article.objects.all().order_by('headline'), [articles[3]])
 
     def test_not_equal_and_equal_operators_behave_as_expected_on_instances(self):
         some_pub_date = datetime(2014, 5, 16, 12, 1)

```
What do you think? Do you have any idea?

Reply:
You can also use `.bulk_create()`.

Reply:
Yes, I tried it but with the `bulk_create()` I got:
`TypeError: Model instances without primary key value are unhashable`